src/polygon.cpp

```diff
@@ -472,10 +517,12 @@ int SKdNodeEdges::AnyEdgeCrossings(Vector a, Vector b, int cnt,
     } else {
         SEdgeLl *sell;
         for(sell = edges; sell; sell = sell->next) {
-            SEdge *se = sell->se;
-            if(se->tag == cnt) continue;
+            SEdge * const se = sell->se;
+            if(se->tag == cnt) {
+                continue;
+            }
             if(se->EdgeCrosses(a, b, pi, spl)) {
-                inters++;
+                ++inters;
```

Comment:
It's not a matter of outsmarting the compiler. Pre-increment is the preferred operation in this context because it is simpler, which makes it more readable/understandable and less prone to errors introduced in the future. Ex: imagine if something like this would need to be done in the future: `const size_t num_inters_this_iteration = inters++;` It's easy to make this kind of bug when post-increment is used habitually. It's a shame that the language wasn't called ++C

Reply:
I don't agree that pre-increment is simpler or more readable in this context (indeed, since no other codebase I work on uses pre-increment pervasively, it trips me up each time), but more importantly, you are not considering the unintended consequences of making such a change globally. Namely:
  * it is not semantics-preserving, and so it will introduce bugs, and we don't have nearly enough test coverage to catch them;
  * it screws up `git blame`, and we don't have nearly enough documentation to not rely on that.

Therefore this change and changes similar to it are not useful, but are harmful.